### DISCUSSION TITLE
Selecting Unicode character using the displayed indices in name mode?

### DISCUSSION THREAD

**Discussion by purxiz** [kovidgoyal/kitty#3217] [Q&A] [answered]
Hello, been using Kitty for a few years now, but just recently have found myself using the unicode input box more and more. 

The documentation on the kitty site implies that I can select the matches for a search in the name (f2) field on the unicode input by typing a "leading period" then typing the index, however I don't seem to be able to figure out exactly what that means.

It's incredibly minor, but my desired functionality is search for a unicode character by name, then be able to select it using it's displayed index, rather than having to go highlight it with the tab and arrow keys.

I'm not sure if that's currently possible with Kitty, the way the docs is written make me think it is, but I just can't figure it out!

EDIT 1 (please don't read just leaving it here for posterity it's wrong)
So I've been poking around `unicode_input/main.py` and it seems like there might actually be a bug here? There is definitely logic to parse words that begin with a '.', so it would seem that typing something like `musical note . 18` should choose the index 18 (base 36? (a-z + 0-9)) of the search musical note. 

On line 322, I found this:
`words = [w for w in words if w != INDEX_CHAR]`

which only omits '.' (or whatever index char) from the search, rather than the entire index char search term. So writing `musical note .18` searches for the literal string `.18` 

EDIT 2: I actually found the bug I think. The conversion from index to unicode character is from base 16 -> base 10. 
Relevant snippet:
```
if index_words:
                    iindex_word = int(index_word.lstrip(INDEX_CHAR), 16)
                    if codepoints and iindex_word < len(codepoints):
                        codepoints = [codepoints[iindex_word]]
```
The functionality actually "works" but in reality, it should be converting from base 36 -> base 10, since the indexing that kitty uses visually is base 36. As an example, you can type in `musical note .18` and get the 24th (base 10) character in the list. However, based on how the characters are labelled visually, you should get the 54th character in the list.

**Comment by kovidgoyal** [comment, marked as answer]
Type a space, period and then index

**Reply by purxiz** [reply]
You can see my edit 2 above, I actually figured it out, the index conversion doesn't match the visually displayed index. Kitty shows you the characters in base 36, but then converts the index you type in from base 16. For example, in the below image, I would expect to type `musical note .o` to get `Musical Note Cluster Notehead` but in reality I have to type `Musical Note .18` because the bases don't match. I'll figure out how to contribute then submit a PR I guess
![image](https://user-images.githubusercontent.com/1489052/103798929-ff20c000-500f-11eb-81f7-808936dfe6bf.png)

**Reply by purxiz** [reply]
Fixed by #3219 hopefully :)
